### PR TITLE
 feat(bpmn-model): enable non-interrupting time cycle boundary events 

### LINKED ISSUE #1702: Validate non interrupting timer cycle events
Add validation rules for non-interrupting timer cycle boundary events.

1. Timer boundary events can now be non-interrupting
1. Timer event in normal flow cannot be a cycle.
1. Interrupting timer event cannot be a cycle.
1. Timer cycle definition conforms to ISO 8601.

Related to #1592

### PR DESCRIPTION
**Pull request by npepinpe**
- allows different support levels for the different types of supported boundary events
- minor refactor of validation: event definition specific validation (context-less) goes into their own validators
- add utility classes to parse repeating time intervals

closes #1702

### CODE REVIEW

**Comment by npepinpe** [comment]
It's debatable whether or not `Interval` and `RepeatingInterval` are needed/in the right place. I couldn't find anything in the standard lib that would do this, or in one of our utilities, but if there exists something I'll gladly take that instead!

I also wasn't sure if putting it in `bpmn-model/` is the right thing, but the module has currently no other dependencies, and I didn't want to add one just for such a small thing...

**Comment by npepinpe** [comment]
Come to think of it, do we actually have a users who want to schedule timers for periods greater than days? e.g. 2 months from now, 1 year from now, etc.

Probably not? Then we can stick with the Duration class and ignore the rest, makes things a lot simpler (i.e. no need to think about time zones, daylight savings time, etc.)

**Review by ChrisKujawa — COMMENTED** [review]
I added some questions maybe we can discuss them f2f

**Comment by npepinpe** [comment]
Pushed changes that should address all feedback points

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/util/time/RepeatingInterval.java [hunk 1]
```diff
@@ -68,25 +68,31 @@ public static RepeatingInterval parse(String text, String intervalDesignator) {
     int repetitions = INFINITE;
     final Interval interval;
 
+    if (text.charAt(0) != 'R') {
```
Comment (Zelldon) [inline bpmn-model/src/main/java/io/zeebe/model/bpmn/util/time/RepeatingInterval.java]: I think there was a misunderstanding :sweat_smile: 

I mean it is cool to have an infinite interval, if I haven't specified `R/`, which means `PT1s` should be possible. But it should not being possible to do the following `X23/PT1s` If I specify a repetition then it should be R and only R. Do you know what I mean? :sweat_smile:

Comment (npepinpe) [reply bpmn-model/src/main/java/io/zeebe/model/bpmn/util/time/RepeatingInterval.java]: Ah, I thought we decided not to have an implicit infinite interval if none is specified. I also thought it was useful, but not important enough that I would insist on keeping it :sweat_smile:

Comment (saig0) [reply bpmn-model/src/main/java/io/zeebe/model/bpmn/util/time/RepeatingInterval.java]: I'm not sure about the first case. I don't think that the "R" is optional. Also, it is not supported in Cam7.

**Review by ChrisKujawa — APPROVED** [review]

**Pull request merged**